api/middleware/auth.js

```diff
@@ -0,0 +1,16 @@
+const jwt = require("jsonwebtoken");
+require("dotenv").config();
+const validateToken = async (request, response, next) => {
+  try {
+    const accessToken = request.headers.authorization;
+    const decode = jwt.verify(accessToken, process.env.SECRET_KEY);
+    request.userId = decode;
```

Comment:
decode를 바로 넘겨주면 안 되고, decode 된 결과 안에서 필요한 값을 reqeust.userId에 넣어주세요.
```suggestion
    const payload = jwt.verify(accessToken, process.env.SECRET_KEY);
    request.userId = payload.userId; // payload.userId에서 userId는 토큰을 만들 때의 값에 따라 달라 질 수 있습니다.
```